scylla/src/transport/topology.rs

```diff
@@ -743,28 +743,30 @@ impl MetadataReader {
     }
 }
 
-async fn query_metadata(
-    conn: &Arc<Connection>,
-    connect_port: u16,
-    keyspace_to_fetch: &[String],
-    fetch_schema: bool,
-) -> Result<Metadata, QueryError> {
-    let peers_query = query_peers(conn, connect_port);
-    let keyspaces_query = query_keyspaces(conn, keyspace_to_fetch, fetch_schema);
+impl ControlConnection {
+    async fn query_metadata(
+        self,
+        connect_port: u16,
+        keyspace_to_fetch: &[String],
+        fetch_schema: bool,
+    ) -> Result<Metadata, QueryError> {
+        let peers_query = self.clone().query_peers(connect_port);
```

Comment:
🌱 When porting the PR to the `main` branch, after applying my previous comment about ControlConnection impls, I think it would be a good idea to extract `control_connection` module to a separate file.

Reply:
I think `ControlConnection` could be extracted to a separate file, but with those higher-level methods (`query_peers` etc.) left in the `metadata.rs` file. The reason is that `ControlConnection` itself is a medium, which should be logically separate from the logic performing queries using it.